google/cloud/storage/emulator/tests/test_gcs.py

```diff
@@ -314,7 +314,7 @@ def test_init_multipart(self):
         request = utils.common.FakeRequest(
             args={},
             headers={"content-type": "multipart/related; boundary=foo_bar_baz"},
-            data=b'--foo_bar_baz\r\nContent-Type: application/json; charset=UTF-8\r\n{"name": "object", "metadata": {"key": "value"}}\r\n--foo_bar_baz\r\nContent-Type: image/jpeg\r\n123456789\r\n--foo_bar_baz--\r\n',
+            data=b'--foo_bar_baz\r\nContent-Type: application/json; charset=UTF-8\r\n\r\n{"name": "object", "metadata": {"key": "value"}}\r\n--foo_bar_baz\r\nContent-Type: image/jpeg\r\n\r\n123456789\r\n--foo_bar_baz--\r\n',
```

Comment:
What's the reason for this change? Will it affect whether request from other clients are acceptable to the emulator?

Reply:
IIRC, the reason why I use `\r\n\r\n` in the first place was because I noticed that sometime `google-cloud-cpp` uses `\r\n` and sometime `\r\n\r\n`. You could see my comment here:

https://github.com/googleapis/google-cloud-cpp/blob/e678e4ee830ff171a4e2ec95070525fc9d4b107c/google/cloud/storage/emulator/utils/common.py#L273

Reply:
Thanks @vnvo2409, I missed that comment. Maybe that's something that can be fixed in the C++ client instead, but the GCS API does allow it if different requests from the C++ client use it.

Reply:
I think (a) each header is required to end with `\r\n`, and (b) there is supposed to be an empty line after all the headers and before the payload.  In other words, the C++ client is *required* (but maybe does not) send a `\r\n\r\n` at the end of the last header.

It seems to do so:

https://github.com/googleapis/google-cloud-cpp/blob/afc9d24e8ceb3a19826e4d2f533950460f3c6a18/google/cloud/storage/internal/curl_client.cc#L1375

Since the tests pass with the new multipart decoder I am inclined to say that whatever bugs we have in the C++ client library can be fixed later.